proarc-common/src/main/java/cz/cas/lib/proarc/common/mods/ndk/NdkMapper.java

```diff
@@ -148,8 +149,13 @@ public ModsDefinition fromJsonObject(ObjectMapper jsMapper, String json, Context
             } else {
                 descriptionStandard.setValue(ModsConstants.VALUE_DESCRIPTIONSTANDARD_AACR);
             }
-            mods.getRecordInfo().get(0).getDescriptionStandard().add(0, descriptionStandard);
-            return mods;
+            if (mods.getRecordInfo().size() == 0) {
```

Comment:
Přidání descriptionStandard v `fromJsonObject()` nastaví hodnotu pro všechny NDK objekty zaslané pouze ve formátu JSON. Formuláře, ale počítají jen s vybranými NDK typy a změna předaná ve formátu XML tímto není řešena vůbec.

Správné místo na doplnění standardu je nejspíš patřičná subclass `NdkMapper` a metoda `createMods()`, která je volána při zápisu pro napársovaný MODS bez ohledu na formát z https://github.com/proarc/proarc/blob/fd6149e39492baf7a3a40169f36e420a218497cc/proarc-common/src/main/java/cz/cas/lib/proarc/common/object/ndk/NdkMetadataHandler.java#L500

Pomohlo by přidání testu.

Reply:
Mohl bys mi podrobněji popsat co jsi myslel prvním odstavcem tohoto komentáře? Chápu, že se hodnota nastavuje jen u objektů (patřících do určité množiny) poslaných ve formátu JSON.  Také chápu, když vyplním popis modelu pomocí XML, tak se mi automaticky nedoplní standard, Nerozumím ale této větě: "Formuláře, ale počítají jen s vybranými NDK typy".

Reply:
> Nerozumím ale této větě: "Formuláře, ale počítají jen s vybranými NDK typy".

`NdkMapper.fromJsonObject` se volá pro větší množinu model IDs, než pro které jsi upravil formuláře. Např. formulář pro `model:ndkarticle` s touto hodnotou nepočítá. Podle mě je tahle změna nežádoucí např. pro  `model:ndkperiodicalvolume`. Metadata ročníku s RDA ani AACR nemají nic společného. Nebo mají? Čekal bych že tohle platí pro tituly převzaté z katalogu nebo vyplněné podle těchto katalogizačních pravidel.

Reply:
Ano NdkMapper.fromJsonObject se volá pro větší množinu modelů. Nicméně na řádce 140 je podmínka, po které se tato změna týká jen vybraných modelů (monografie, příloha monografie, periodikum, příloha periodika, hudebnina a kartografie). Ostatní modely, neprojdou podmínkou.

Reply:
Už vzhledem k tomu, že metoda fromJsonObject je virtuální (=je možné ji overridnout), tak je čistější to mít přímo v těch konkrétních subclass.

Reply:
Rozumím, zítra opravím